basic-loadbalancer-aware-server/src/main/java/com/rtbhouse/grpc/loadbalancer/LoadBalancerConnector.java

```diff
@@ -0,0 +1,171 @@
+package com.rtbhouse.grpc.loadbalancer;
+
+import com.spotify.dns.DnsSrvResolver;
+import com.spotify.dns.DnsSrvResolvers;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Handles backend server communication with load balancer (as defined in signup.proto). You should
+ * provide server's address, port and domain addresses (host:port) of services that server serves.
+ * These names have to be the same as those used by clients when creating gRPC channels, e.g.
+ * through ManagedChannelBuilder.forAddress(host, port). You should also provide a list of
+ * loadbalancers' addresses or a domain address to resolve them automatically, by checking SRV
+ * records for the domain, just as clients do.
+ *
+ * <p>There is also serverWeight, which is optional. You may use it to modify the percentage of
+ * requests that server receives. Default value is 100, and it means that server should receive c.a.
+ * 1/N of all requests, where N is the total number of backend servers. Generally, load balancer
+ * sums all servers' weights, and every server gets proportional (server_weight/sum) number of
+ * requests.
+ *
+ * <p>You can pause/resume, or stop/start the connector multiple times. Notice, that using
+ * pause()/resume() is less expensive than stop()/start(), because gRPC channel is not being closed
+ * then. When there is no state change needed (e.g. when you call resume() on a working connector),
+ * pause() and resume() won't be synchronizing. Start()/stop() are always synchronizing.
+ */
+public class LoadBalancerConnector {
+  public static final int WEIGHT_NOT_SET = -1;
+  public static final String GRPCLB_DNS_PREFIX = "_grpclb._tcp.";
+  private final List<SingleLoadBalancerConnector> lbConnectors;
+  private volatile ConnectorStatus status;
+
+  private LoadBalancerConnector(
+      int serverPort,
+      InetAddress serverAddress,
+      String[] services,
+      List<Pair<String, Integer>> s,
```

Comment:
Zamiast tej pary zrobiłbym w tej klasie prywatną statyczną klasę i jej używał np:
`private static class Address {
    String hostName;
    Integer port;
}
`
Można by w niej też zaimplementować to parsowanie ze String z drukropkiem.

Do tego tę nazwę s bym zmienił na coś bardziej znaczącego